src/core/simulation.jl

```diff
@@ -736,32 +722,32 @@ function initial_condition_update!(
             ic,
             quantity,
             previous_value,
-            get_number(stage),
+            get_simulation_number(problem),
```

Comment:
I'm not sure that this is intuitive.

Reply:
Do you suggest reverting to get_number? The idea was to indicate this method is used in a simulation.